core/src/json_schema.rs

```diff
@@ -112,7 +218,6 @@ pub mod simplified {
     // simplified::Schema to disallow nested AnyOfs.
     #[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone)]
     pub enum Atomic {
-        Any,
```

Comment:
There's no reason to treat this separately from just allowing `BsonTypeName::Any`. I initially was converting `Scalar(Any)` to `Any`, and realized there is no advantage to a top level `Any` in this case (still makes sense in mongosql, I think, for brevity).